### PR TITLE
Deprecate BasicAuthSoapClient. Update Curl.php to check for auth type…

### PR DESCRIPTION
**Pull request by melmccann**
… Basic and add constants for auth types.

### CODE REVIEW

File: src/BeSimple/SoapClient/Curl.php [hunk 1]
```diff
@@ -111,7 +115,9 @@ public function __construct(array $options = array(), $followLocationMaxRedirect
                 isset($options['preemptive_auth']) &&
                 true === $options['preemptive_auth']
             ) {
-                $headers[] = sprintf('Authorization: Basic %s', base64_encode($curlUserPwd));
+                if($options['auth_type'] === self::AUTH_TYPE_BASIC) {
```
[the remark this answers is not in the document]
Comment (melmccann) [reply src/BeSimple/SoapClient/Curl.php]: I will but to me it just reads weird that way :-D
I will add auth_type in the yaml

File: src/BeSimple/SoapClient/Curl.php [hunk 2]
```diff
@@ -111,7 +115,9 @@ public function __construct(array $options = array(), $followLocationMaxRedirect
...
+                    $headers[] = sprintf('Authorization: Basic %s', base64_encode($curlUserPwd));
+                }
```
[the remark this answers is not in the document]
Comment (melmccann) [reply src/BeSimple/SoapClient/Curl.php]: talk afk ha ha

File: src/BeSimple/SoapClient/SoapClient.php [hunk 3]
```diff
@@ -282,10 +282,9 @@ protected function __doRequest2(SoapRequest $soapRequest)
      */
     protected function filterRequestHeaders(SoapRequest $soapRequest, array $headers)
     {
-        if (isset($this->_login) && isset($this->_password) && Curl::AUTH_TYPE_NONE === $this->authType && Curl::AUTH_TYPE_NTLM !== $this->authType) {
+        if (isset($this->_login) && isset($this->_password) && Curl::AUTH_TYPE_BASIC === $this->authType) {
```
Comment (andreskrey) [inline src/BeSimple/SoapClient/SoapClient.php]: Can we reverse this to `Curl::AUTH_TYPE_BASIC === $this->authType && isset($this->_login) && isset($this->_password)` so if it's not BASIC we don't bother checking for the variables?

Comment (melmccann) [reply src/BeSimple/SoapClient/SoapClient.php]: sure... I will keep you happy

File: src/BeSimple/SoapClient/Curl.php [hunk 4]
```diff
@@ -100,19 +104,16 @@ public function __construct(array $options = array(), $followLocationMaxRedirect
...
-            // use preemptive authentication
-            if (
-                isset($options['preemptive_auth']) &&
-                true === $options['preemptive_auth']
-            ) {
-                $headers[] = sprintf('Authorization: Basic %s', base64_encode($curlUserPwd));
-            }
+
+                // use preemptive authentication
+                if (self::AUTH_TYPE_BASIC === $options['auth_type']) {
```
Comment (andreskrey) [inline src/BeSimple/SoapClient/Curl.php]: Can we reuse $authType here?

Comment (melmccann) [reply src/BeSimple/SoapClient/Curl.php]: yes.. I will do for the ticket  @andreymor is reviewing.

File: src/BeSimple/SoapClient/SoapClient.php [hunk 5]
```diff
@@ -452,26 +456,39 @@ protected function loadWsdl($wsdl, array $options)
...
+        foreach ($this->requestHeaders as $key => $value) {
+            $requestHeadersStringArray[] = $key.': '.$value;
         }
+
         return $requestHeadersStringArray;
     }
 
-}
\ No newline at end of file
+    public function getAuthType()
```
[the remark this answers is not in the document]
Comment (melmccann) [reply src/BeSimple/SoapClient/SoapClient.php]: There was some strange issue with setting the type here when running the compiler pass in the framework bundle.

File: src/BeSimple/SoapClient/SoapClient.php [hunk 6]
```diff
@@ -104,17 +103,20 @@ class SoapClient extends \SoapClient
      */
     protected $soapKernel = null;
 
+    protected $authType = Curl::AUTH_TYPE_NONE;
```
[the remark this answers is not in the document]
Comment (melmccann) [reply src/BeSimple/SoapClient/SoapClient.php]: There was some strange issue with setting the type here when running the compiler pass in the framework bundle.

**Pull request merged**